File: server/router/cloudpods/compute/servers.go
package compute

import (
	v1 "ygpt/server/api/v1"

	"github.com/gin-gonic/gin"
)

type ServersRouter struct {
}

func (s *ServersRouter) InitServersRouter(Router *gin.RouterGroup) {
	dataRouter := Router.Group("cloud/compute/servers")
	ipApi := v1.ApiGroupApp.CloudApiGroup.ComputeServersApi
	{
		dataRouter.POST("getServerList", ipApi.GetServerList)       // 获取云主机列表
		dataRouter.POST("getWebConsoleSSH", ipApi.GetWebConsoleSSH) // 获取WebSSH连接
		dataRouter.POST("hostShutDown", ipApi.HostShutDown)         // 关机
		dataRouter.POST("hostStart", ipApi.HostStart)               // 开机
		dataRouter.POST("hostRestart", ipApi.HostRestart)           // 重启
		dataRouter.POST("hostSuspend", ipApi.HostSuspend)           // 挂起
		dataRouter.POST("hostResume", ipApi.HostResume)             // 恢复
		dataRouter.POST("hostCreate", ipApi.HostCreate)             // 云主机创建
		dataRouter.POST("hostDelete", ipApi.HostDelete)             // 删除
		dataRouter.POST("hostDeleteByIds", ipApi.HostDeleteByIds)   // 批量删除
		dataRouter.POST("hostDetails", ipApi.HostDetails)           // 获取主机详情
		dataRouter.POST("getGpuInfo", ipApi.GetGpuInfo)             // 获取GPU信息
		dataRouter.POST("manageGpu", ipApi.ManageGpu)               // 管理GPU
		dataRouter.POST("migrateForecast", ipApi.MigrateForecast)   // 获取云主机可迁移的宿主机列表
		dataRouter.POST("migrate", ipApi.Migrate)                   // 云主机迁移

		dataRouter.POST("getBaremetalList", ipApi.GetBareServerList) // 获取裸金属列表
		dataRouter.POST("baremetalCreate", ipApi.BaremetalCreate)    //裸金属创建

		dataRouter.POST("queryCpu", ipApi.GetHostMonitorCPUData)                   // 获取监控CPU数据
		dataRouter.POST("queryMem", ipApi.GetHostMonitorMemData)                   // 获取监控内存数据
		dataRouter.POST("queryDisk", ipApi.GetHostMonitorDiskData)                 // 获取监控磁盘使用率
		dataRouter.POST("queryDiskRead", ipApi.GetHostMonitorDiskReadData)         // 获取监控磁盘读速率
		dataRouter.POST("queryDiskWrite", ipApi.GetHostMonitorDiskWriteData)       // 获取监控磁盘写速率
		dataRouter.POST("queryBpsRecv", ipApi.GetHostMonitorBpsRecvData)           // 获取监控网络入带宽
		dataRouter.POST("queryBpsSent", ipApi.GetHostMonitorBpsSentData)           // 获取监控网络出带宽
		dataRouter.POST("queryPpsRecv", ipApi.GetHostMonitorPpsRecvData)           // 获取监控网络收包数
		dataRouter.POST("queryPpsSent", ipApi.GetHostMonitorPpsSentData)           // 获取监控网络发包数
		dataRouter.POST("queryGpu", ipApi.GetHostMonitorGPUData)                   // 获取GPU使用率
		dataRouter.POST("queryGpuMem", ipApi.GetHostMonitorGPUMemData)             // 获取GPU显存使用率
		dataRouter.POST("queryGpuTemperature", ipApi.GetHostMonitorGpuTemperature) // 获取Gpu温度

		dataRouter.POST("getResSkuList", ipApi.GetResSkuList) // 查询套餐列表
		dataRouter.POST("getLoginInfo", ipApi.GetLoginInfo)   // 获取账号密码
		dataRouter.POST("resetPassword", ipApi.ResetPassword) // 重置账号密码

		dataRouter.POST("getGpuList", ipApi.GetGpuList)                       // 获取可使用的GPU信息
		dataRouter.POST("getRecycleServerList", ipApi.GetRecycleServerList)   // 获取回收站列表
		dataRouter.POST("clearRecycleHost", ipApi.ClearRecycleHost)           // 清除回收主机
		dataRouter.POST("clearRecycleHostByIds", ipApi.ClearRecycleHostByIds) // 清除回收主机
		dataRouter.POST("recoverRecycleHost", ipApi.RecoverRecycleHost)       // 恢复回收主机

		dataRouter.POST("updateName", ipApi.UpdateName) // 修改名称

		dataRouter.POST("forecastAgent", ipApi.ForecastAgent)                     //探测是否有agent
		dataRouter.POST("forecastAgentCanInstall", ipApi.ForecastAgentCanInstall) //探测是否具备安装agent的条件
		dataRouter.POST("setAgentCanInstall", ipApi.SetAgentCanInstall)           //输入ssh信息设置其能安装agent
		dataRouter.POST("installAgent", ipApi.InstallAgent)                       //安装agent
	}
}

File: web/src/api/cloudpods/compute/servers.js
import request from '@/utils/request'

//恢复回收主机
export function recoverRecycleHost(params) {
    return request({
        url: 'cloud/compute/servers/recoverRecycleHost',
        method: 'post',
        data: params
    })
}
//清除回收主机
export function clearRecycleHost(params) {
    return request({
        url: 'cloud/compute/servers/clearRecycleHost',
        method: 'post',
        data: params
    })
}
//批量清除回收主机
export function clearRecycleHostByIds(params) {
    return request({
        url: 'cloud/compute/servers/clearRecycleHostByIds',
        method: 'post',
        data: params
    })
}
//获取回收站列表
export function getRecycleServerList(params) {
    return request({
        url: 'cloud/compute/servers/getRecycleServerList',
        method: 'post',
        data: params
    })
}

//获取可使用的GPU信息
export function getGpuList() {
    return request({
        url: 'cloud/compute/servers/getGpuList',
        method: 'post',
    })
}

//云主机列表
export function getHostList(query) {
    return request({
        url: 'cloud/compute/servers/getServerList',
        method: 'post',
        data: query
    })
}
//获取裸金属列表
export function getBaremetalList(query) {
    return request({
        url: 'cloud/compute/servers/getBaremetalList',
        method: 'post',
        data: query
    })
}
export  function  getWebSSHConn(params){
    return request({
        url:'cloud/compute/servers/getWebConsoleSSH',
        method:'post',
        data: params
    })
}
//删除
export  function  setHostDelete(params){
    return request({
        url:'cloud/compute/servers/hostDelete',
        method:'post',
        data: params
    })
}
//关机
export  function  setHostShutDown(params){
    return request({
        url:'cloud/compute/servers/hostShutDown',
        method:'post',
        data: params
    })
}
// 开机
export  function  setHostStart(params){
    return request({
        url:'cloud/compute/servers/hostStart',
        method:'post',
        data: params
    })
}
//重启
export  function  setHostRestart(params){
    return request({
        url:'cloud/compute/servers/hostRestart',
        method:'post',
        data: params
    })
}
//挂起
export  function  setHostSuspend(params){
    return request({
        url:'cloud/compute/servers/hostSuspend',
        method:'post',
        data: params
    })
}
//删除
export  function  deleteHost(params){
    return request({
        url:'cloud/compute/servers/hostDelete',
        method:'post',
        data: params
    })
}
//删除
export  function  deleteHostByIds(params){
    return request({
        url:'cloud/compute/servers/hostDeleteByIds',
        method:'post',
        data: params
    })
}
//恢复
export  function  setHostResume(params){
    return request({
        url:'cloud/compute/servers/hostResume',
        method:'post',
        data: params
    })
}
//创建
export  function  addHost(params){
    return request({
        url:'cloud/compute/servers/hostCreate',
        method:'post',
        data: params
    })
}

//查询套餐列表
export  function  getResSkuList(params){
    return request({
        url:'cloud/compute/servers/getResSkuList',
        method:'post',
        data: params
    })
}
//获取账号密码
export  function getLoginInfo(params){
    return request({
        url:'cloud/compute/servers/getLoginInfo',
        method:'post',
        data: params
    })
}
//获取账号密码
export  function resetPassword(params){
    return request({
        url:'cloud/compute/servers/resetPassword',
        method:'post',
        data: params
    })
}

// 裸金属创建
export function createBakemeats(data){
    return request({
        url:'cloud/compute/servers/baremetalCreate',
        method:'post',
        data:data
    })
}
// 修改名称
export function updateName(data){
    return request({
        url:'cloud/compute/servers/updateName',
        method:'post',
        data:data
    })
}
// 获取实例详情
export function hostDetails(data){
    return request({
        url:'cloud/compute/servers/hostDetails',
        method:'post',
        data:data
    })
}
//获取CPU监控数据
export  function  getMonitorCpuData(params){
    return request({
        url:'cloud/compute/servers/queryCpu',
        method:'post',
        data: params
    })
}
//获取内存监控数据
export  function  getMonitorMemData(params){
    return request({
        url:'cloud/compute/servers/queryMem',
        method:'post',
        data: params
    })
}
//获取磁盘监控数据
export  function  getMonitorDiskData(params){
    return request({
        url:'cloud/compute/servers/queryDisk',
        method:'post',
        data: params
    })
}
//获取磁盘读速率监控数据
export  function  getMonitorDiskReadData(params){
    return request({
        url:'cloud/compute/servers/queryDiskRead',
        method:'post',
        data: params
    })
}
//获取磁盘写速率监控数据
export  function  getMonitorDiskWriteData(params){
    return request({
        url:'cloud/compute/servers/queryDiskWrite',
        method:'post',
        data: params
    })
}
//获取网络入带宽监控数据
export  function  getMonitorBpsRecvData(params){
    return request({
        url:'cloud/compute/servers/queryBpsRecv',
        method:'post',
        data: params
    })
}
//获取网络出带宽监控数据
export  function  getMonitorBpsSentData(params){
    return request({
        url:'cloud/compute/servers/queryBpsSent',
        method:'post',
        data: params
    })
}
//获取网络收包数监控数据
export  function  getMonitorPpsRecvData(params){
    return request({
        url:'cloud/compute/servers/queryPpsRecv',
        method:'post',
        data: params
    })
}
//获取网络发包数监控数据
export  function  getMonitorPpsSentData(params){
    return request({
        url:'cloud/compute/servers/queryPpsSent',
        method:'post',
        data: params
    })
}
//获取GPU使用率
export  function  getMonitorGpuData(params){
    return request({
        url:'cloud/compute/servers/queryGpu',
        method:'post',
        data: params
    })
}
//获取GPU显存使用率
export  function  getMonitorGpuMemData(params){
    return request({
        url:'cloud/compute/servers/queryGpuMem',
        method:'post',
        data: params
    })
}
//获取Gpu温度
export  function  getMonitorGpuTemperatureData(params){
    return request({
        url:'cloud/compute/servers/queryGpuTemperature',
        method:'post',
        data: params
    })
}
//获取云主机可迁移的宿主机列表
export  function  getMigrateForecastHost(params){
    return request({
        url:'cloud/compute/servers/migrateForecast',
        method:'post',
        data: params
    })
}
//迁移
export  function  doMigrate(params){
    return request({
        url:'cloud/compute/servers/migrate',
        method:'post',
        data: params
    })
}
//探测是否有agent
export function forecastAgent(params){
    return request({
        url:'cloud/compute/servers/forecastAgent',
        method:'post',
        data: params
    })
}
//探测是否具备安装agent的条件
export function forecastAgentCanInstall(params){
    return request({
        url:'cloud/compute/servers/forecastAgentCanInstall',
        method:'post',
        data: params
    })
}
//输入ssh信息设置其能安装agent
export function setAgentCanInstall(params){
    return request({
        url:'cloud/compute/servers/setAgentCanInstall',
        method:'post',
        data: params
    })
}
//安装agent
export function installAgent(params){
    return request({
        url:'cloud/compute/servers/installAgent',
        method:'post',
        data: params
    })
}
//获取gpu信息列表
export function getGpuInfo(params){
    return request({
        url:'cloud/compute/servers/getGpuInfo',
        method:'post',
        data: params
    })
}
//获取gpu信息列表
export function manageGpu(params){
    return request({
        url:'cloud/compute/servers/manageGpu',
        method:'post',
        data: params
    })
}